src/core/crypto/index.ts

```diff
@@ -1,11 +1,16 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-export * from "./asymmetricCrypto";
+import { Ed25519PublicKey } from "./ed25519";
+import { MultiEd25519PublicKey } from "./multiEd25519";
+import { MultiKey } from "./multiKey";
+import { AnyPublicKey } from "./singleKey";
+
 export * from "./ed25519";
+export * from "./hdKey";
 export * from "./multiEd25519";
-export * from "./secp256k1";
 export * from "./multiKey";
-export * from "./hdKey";
-export * from "./anyPublicKey";
-export * from "./anySignature";
+export * from "./secp256k1";
+export * from "./singleKey";
+
+export type PublicKeyInput = Ed25519PublicKey | MultiEd25519PublicKey | AnyPublicKey | MultiKey;
```

Comment:
isnt AnyPublicKey contain MultiKey already?

Reply:
Mm unless I got it wrong, `AnyPublicKey` and `AnySignature` are wrappers for single public keys and signatures.
`MultiKey` will take as an input N instances of single keys, and `MultiKeySignature` (introduced in this PR) will take as an input N instances of single keys

Reply:
So the set of all possible "accepted" public keys are:
- Ed25519 (legacy)
- MultiEd25519 (legacy)
- AnyPublicKey aka SingleKey
- MultiKey

Reply:
notably, you can't use a Secp256k1PublicKey directly